src/relayedTransactionV2Builder.ts

```diff
@@ -0,0 +1,59 @@
+import {Transaction} from "./transaction";
+import {TransactionPayload} from "./transactionPayload";
+import {AddressValue, BytesValue, ContractFunction, U64Value} from "./smartcontracts";
+import {IAddress, IChainID, IGasLimit, INonce, ITransactionPayload} from "./interface";
+import {ISigner} from "@elrondnetwork/erdjs-walletcore/out/interface";
+import {INetworkConfig} from "./interfaceOfNetwork";
+
+export class RelayedTransactionV2Builder {
```

Comment:
Allt tx builders in this repo are defined here: https://github.com/ElrondNetwork/elrond-sdk-erdjs/blob/main/src/transactionFactory.ts. Could we move the logic here to keep consistency?

Reply:
it is rather more similar to `tokenTransferBuilders.ts` so I will leave it as it is

Reply:
`tokenTransferBuilders` builds the payLoad. `transactionFactory` builds the txs. You could've split `relayedTransactionV2Builder` in 3 to keep consistency:
- gasEstimator for relayed tx
- payloadBuilder for relayed tx
- txFactory to create the relayed tx.

Again, let's also hear @andreibancioiu 's opinion

Reply:
The need for _consistency_ is indeed a very good argument. On the other hand, this new component is self-contained, which is also good.

If we were to move `gasEstimator.ts`, `tokenTransferBuilders.ts` and `transactionFactory.ts` to a new package called `transfers` (in the future), then this design is a good choice. It's possible that in the future we might even extract the following logic outside of erdjs (and move it to separate packages):
 - transfers (EGLD and all kinds of ESDT)
 - relayed transactions (V1 & V2)
 - staking & delegation SC calls (actually, missing here now)
 - esdt management calls (actually, missing here now)

So, no strong opinion at this moment (can stay as it is if we desire to further split erdjs functionality to separate NPM packages),